examples/hello-world/src/test-api.js

```diff
@@ -45,22 +45,22 @@ export default class TestApi extends FreshDataApi {
 		},
 	}
 
-	static endpoints = {
-		greetings: {
-			read: ( methods, endpointPath, params ) => {
-				const { get } = methods;
-				const fullEndpointPath = [ 'greetings', ...endpointPath ];
-				const value = get( fullEndpointPath )( params );
-				return value;
-			},
-		},
+	static operations = {
+		read: ( { get } ) => ( resourceNames ) => {
+			return resourceNames.reduce( ( requests, name ) => {
+				if ( 'greetings' === name ) {
+					requests[ name ] = get( [ name ] )();
+				}
+				return requests;
+			}, {} );
```

Comment:
what is the purpose of this function? it looks like it's establishing a library of data types that it will then `get` with some passed-in getter, but that getter takes an id as input. if we call this with a `get` parameter anyway, why is this needed? could we not just call `get( 'my-thing' )`?

Reply:
The `get` passed in here is an API method, which is a general-purpose API function. The WP REST API only has two methods: `GET` and `POST`. They are implemented once and passed in to every operation to be used.

This provides an abstraction point where the methods can be easily implemented in different ways on different environments (such as using a proxy in Calypso on wordpress.com, or using OAuth authenticated HTTP requests, or using `apiRequest` from `@wordpress/data`. The methods also could be pointed to another data system for further integration.

Reply:
>  could we not just call get( 'my-thing' )?

I think my longer explanation [above](https://github.com/coderkevin/fresh-data/pull/56/files/3aeeb9ca840ceea73f3a88b263d37986fe931630#r200397801) explains why one would want to do resource-specific things here in the operations, rather than the methods. Let me know if you still have questions.